### PR TITLE
fix(ui): team options on owner widget

### PR DESCRIPTION
**Pull request by chirag-madlani**
### Describe your changes:

- fixed https://github.com/open-metadata/OpenMetadata/issues/10543#issuecomment-1505621009

https://user-images.githubusercontent.com/12962843/234527656-3a22e31e-65b3-4ca8-a21c-e2d81c5b3e6a.mov

I worked on ... because ...

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/common/SelectableList/SelectableList.component.tsx [hunk 1]
```diff
@@ -114,19 +114,25 @@ export const SelectableList = ({
...
 
-      setUniqueOptions((prevData) => [...prevData, ...data]);
-      setPagingInfo(paging);
-    }
-  };
+        setUniqueOptions((prevData) => [...prevData, ...data]);
+        setPagingInfo(paging);
+      }
+    },
+    [pagingInfo]
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/SelectableList/SelectableList.component.tsx:134]:
```suggestion
    [pagingInfo,uniqueOptions,searchText]
```

File: openmetadata-ui/src/main/resources/ui/src/components/common/SelectableList/SelectableList.component.tsx [hunk 2]
```diff
@@ -114,19 +114,25 @@ export const SelectableList = ({
...
-    ) {
-      const { data, paging } = await fetchOptions(searchText, pagingInfo.after);
+  const onScroll: UIEventHandler<HTMLElement> = useCallback(
+    async (e) => {
+      if (
+        e.currentTarget.scrollHeight - e.currentTarget.scrollTop ===
+          ADD_USER_CONTAINER_HEIGHT &&
+        pagingInfo.after &&
+        uniqueOptions.length < pagingInfo.total
+      ) {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/SelectableList/SelectableList.component.tsx:124]: I would recommend adding some comments here.

File: openmetadata-ui/src/main/resources/ui/src/components/common/UserTeamSelectableList/UserTeamSelectableList.component.tsx [hunk 3]
```diff
@@ -119,15 +120,21 @@ export const UserTeamSelectableList = ({
...
+        };
       } catch (error) {
         return { data: [], paging: { total: 0 } };
       }
     } else {
       try {
         const { data } = await searchData(
           WILD_CARD_CHAR,
-          isNumber(after) ? after : 1,
+          isNaN(afterPage) ? 1 : afterPage,
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/UserTeamSelectableList/UserTeamSelectableList.component.tsx:137]: We are already checking for isNaN above, so we don't need to check it again here.

```js
const afterPage = isNaN(Number(after)) ? 1 : Number(after);
```

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

**Review by shahsank3t — APPROVED** [review]

**Pull request merged**